onnxruntime/test/providers/coreml/coreml_basic_test.cc

```diff
@@ -257,6 +257,98 @@ TEST(CoreMLExecutionProviderTest, TestNameSanitization) {
   // TensorRT does not support Clip opset 11 yet.
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
 }
+
+TEST(CoreMLExecutionProviderTest, TestBinaryFp16) {
```

Comment:
Do we need to add separate tests here or could we update onnxruntime\test\providers\cpu\math\element_wise_ops_test.cc to run the MLFloat16 tests it has for CoreML? 

We're also going to add some xnnpack fp16 kernels so the more common test code that is used the better.

Reply:
Good idea. Refactored some of elemenwise UTs. It's still a bit messy for FP16/Bf16 test.